src/controllers/open-data-datasets.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * Gets the datasets from the cache and returns as a response
+ */
+module.exports = async function(req, res) {
+  let cache = req.cache
+  let datasets
+  try {
+    datasets = await cache.getDatasetData()
```

Comment:
Add some logic here for determining if the keys are in the cache. If they are, return it just like you do here. If they aren't, do your dataset pulling here.

Reply:
Logic can be found [here](83/files#diff-d805a1d29385626e20c504f74944b94eR40).